File: tests/es6shim/src/test/java/org/treblereel/j2cl/shim/ES6TransitiveTest.js
import {ES6Test} from "./ES6Test.js";
import {ES6TestZZ} from "./ES6TestZZ.js";

class ES6TransitiveTest {

    constructor() {
        this.es6Test = new ES6Test();
        this.es6Test2 = new ES6TestZZ();
    }

    /**
    * @return {string}
    */
    getEs6Testid() {
        return this.es6Test.id;
    }

        /**
        * @return {string}
        */
        getEs6Test2id() {
            return this.es6Test2.id;
        }

}

export { ES6TransitiveTest };

File: tests/es6shim/src/test/java/org/treblereel/j2cl/shim/ES6TransitiveTest.java
package org.treblereel.j2cl.shim;

import jsinterop.annotations.JsType;
import org.treblereel.j2cl.processors.annotations.ES6Module;

/**
 * @author Dmitrii Tikhomirov
 * Created by treblereel 1/27/21
 */
@ES6Module
@JsType(isNative = true, namespace = "org.treblereel.j2cl.shim")
public class ES6TransitiveTest {

    private ES6Test es6Test;
    private ES6Test2 es6Test2;

    public native boolean getEs6Testid();

    public native boolean getEs6Test2id();
}
